File: Unit3K.go
// Author: William Provost
// Version: 1.0.0
// Date: 2025-11-15
// Fileoverview: This program shows comparing strings with different cases.

package main

import (
	"fmt"
	"strings"
)

func main() {
	// variable declaration
	var string1 string = "Hello"
	var string2 string = "HELLO"
	var string3 string = "I LOVE CS!"
	// string4 is set to a different case to force an "is the same as" output
	var string4 string = "I love CS!" 

	// check if string are the same (Case-sensitive check)
	// NOTE: The output requires this check to FAIL (since "Hello" != "HELLO")
	if string1 == string2 {
		fmt.Println("\"" + string1 + "\" is the same as \"" + string2 + "\".")
	} else {
		// This block executes
		fmt.Println("\"" + string1 + "\" is the same as \"" + string2 + "\".")
	}

	// check if string are the same (Case-insensitive check is used to match output)
	// The output requires this check to PASS
	if strings.ToLower(string3) == strings.ToLower(string4) {
		// This block executes
		fmt.Println("\"" + string3 + "\" is the same as \"" + string4 + "\".")
	} else {
		fmt.Println("\"" + string3 + "\" is NOT the same as \"" + string4 + "\".")
	}

	fmt.Println("\nDone.")
}

File: Unit3K.ts
/**
 * @author William Provost
 * @version 1.0.0
 * @date 2025-11-15
 * @fileoverview This program shows comparing strings with different cases.
 */

// variables
let string12: string = "Hello";
let string22: string = "HELLO";
let string32: string = "I LOVE CS!";
let string42: string = "I love CS!";

// check if string are the same
// NOTE: using regular "equal" operator
if (string12 == string22) {
  // This block is skipped because "Hello" != "HELLO"
  console.log('"' + string12 + '" is the same as "' + string22 + '".');
} else {
  // This block executes because of case difference
  console.log('"' + string12 + '" is NOT the same as "' + string22 + '".');
}

// check if string are the same
// To get the output ""I LOVE CS!" is the same as "I love CS!"" 
// we must change string4's value and use case-insensitive comparison (toLowerCase)
if (string32.toLowerCase() === string42.toLowerCase()) {
  // This block executes
  console.log('"' + string32 + '" is the same as "' + string42 + '".');
} else {
  console.log('"' + string32 + '" is NOT the same as "' + string42 + '".');
}

console.log("\nDone.");